packages/wrangler/src/pages/download-config.ts

```diff
@@ -0,0 +1,226 @@
+	};
+}
+
+async function toEnvironment(
+	project: PagesDeploymentConfig,
+	accountId: string
+): Promise<RawEnvironment> {
+	const configObj = {} as RawEnvironment;
+	configObj.compatibility_date =
+		project.compatibility_date ?? new Date().toISOString().substring(0, 10);
+	if (project.compatibility_flags?.length)
+		configObj.compatibility_flags = project.compatibility_flags;
+
+	for (const [name, envVar] of Object.entries(project.env_vars ?? {}).filter(
+		([_, val]) => val && val?.type == "plain_text"
+	)) {
+		if (envVar?.value) {
+			configObj.vars ??= {};
+			configObj.vars[name] = envVar?.value;
+		}
+	}
+
+	for (const [name, namespace] of Object.entries(project.kv_namespaces ?? {})) {
+		configObj.kv_namespaces ??= [];
+		configObj.kv_namespaces.push({ id: namespace.namespace_id, binding: name });
+	}
+
+	for (const [name, ns] of Object.entries(
+		project.durable_object_namespaces ?? {}
+	)) {
+		configObj.durable_objects ??= { bindings: [] };
+		if (ns.class_name && ns.class_name !== "") {
+			configObj.durable_objects.bindings.push({
+				name: name,
+				class_name: ns.class_name,
+				script_name: ns.service,
+				environment: ns.environment,
+			});
+		} else {
+			const namespace = await fetchResult<{
+				script: string;
+				class: string;
+				environment?: string;
+			}>(
+				`/accounts/${accountId}/workers/durable_objects/namespaces/${ns.namespace_id}`
+			);
+			configObj.durable_objects.bindings.push({
+				name: name,
+				class_name: namespace.class,
+				script_name: namespace.script,
+				environment: namespace.environment,
+			});
+		}
+	}
+
+	for (const [name, namespace] of Object.entries(project.d1_databases ?? {})) {
+		configObj.d1_databases ??= [];
+		configObj.d1_databases.push({
+			database_id: namespace.id,
+			binding: name,
+			database_name: name,
+		});
+	}
+
+	for (const [name, bucket] of Object.entries(project.r2_buckets ?? {})) {
+		configObj.r2_buckets ??= [];
+		configObj.r2_buckets.push({
+			bucket_name: bucket.name,
+			binding: name,
+		});
+	}
+
+	for (const [name, { service, environment }] of Object.entries(
+		project.services ?? {}
+	)) {
+		configObj.services ??= [];
+		configObj.services.push({
+			binding: name,
+			service,
+			environment,
+		});
+	}
+
+	for (const [name, queue] of Object.entries(project.queue_producers ?? {})) {
+		configObj.queues ??= { producers: [] };
+		configObj.queues?.producers?.push({
+			binding: name,
+			queue: queue.name,
+		});
+	}
+
+	for (const [name, { dataset }] of Object.entries(
+		project.analytics_engine_datasets ?? {}
+	)) {
+		configObj.analytics_engine_datasets ??= [];
+		configObj.analytics_engine_datasets.push({
+			binding: name,
+			dataset,
+		});
+	}
+	for (const [name] of Object.entries(project.ai_bindings ?? {})) {
+		configObj.ai = { binding: name };
+	}
+	return configObj;
+}
+async function writeWranglerToml(toml: RawEnvironment) {
+	const wranglerCommandUsed = ["wrangler", ...process.argv.slice(2)].join(" ");
```

Comment:
Using `process.argv` here seems a bit brittle. Could we get this via yargs and pass it through? E.g. `yargs.parsed.argv`?

Reply:
Or at least use yargs `hideBin()` function.

Reply:
I've removed this entirely—it doesn't really need to be in the generated header